torch/csrc/api/include/torch/nn/options/loss.h

```diff
@@ -27,5 +27,27 @@ struct TORCH_API HingeEmbeddingLossOptions {
   TORCH_ARG(Reduction::Reduction, reduction) = Reduction::Mean;
 };
 
+// ============================================================================
+
+/// Options for a Multi-Margin loss functional and module.
+struct TORCH_API MultiMarginLossOptions {
+  /// Has a default value of :math:`1`. :math:`1` and :math:`2`
+  /// are the only supported values.
+  TORCH_ARG(int64_t, p) = 1;
+  /// Has a default value of :math:`1`.
+  TORCH_ARG(double, margin) = 1.0;
+  /// A manual rescaling weight given to each
+  /// class. If given, it has to be a Tensor of size `C`. Otherwise, it is
+  /// treated as if having all ones.
+  TORCH_ARG(c10::optional<Tensor>, weight) = c10::nullopt;
```

Comment:
My sincere apologies for the change here: after putting more thoughts into how we should represent an optional tensor in C++ API, I came to realize that we should use this instead:
```suggestion
  TORCH_ARG(Tensor, weight) = Tensor();
```
which would be consistent with the representation in other parts of C++ API such as custom autograd functions.
If we do this, then we won't need to do `options.weight().value()` anymore, and can just do `options.weight()`.